src/ReadLine.Demo/Program.cs

```diff
@@ -15,10 +15,19 @@ public static void Main(string[] args)
 
             ReadLine.AutoCompletionHandler = (t, s) =>
             {
-                if (t.StartsWith("git"))
-                    return new string[] { "init", "clone", "pull", "push" };
-                else
-                    return null;
+                
+                string _command = t.Substring(0, t.Length);
+                switch(_command)
+                {
+                    case "git":
+                        return new string[] { "git init", "git clone", "git pull", "git push", "git add" };
+                    case "ls":
+                        return new string[] { "ls -a", "ls -al"};
+                    case "dotnet":
+                        return new string[] { "dotnet restore", "dotnet run", "dotnet new" };
+                    default:
+                        return null;
+                }
```

Comment:
Did you test this? The point of this block when I wrote it was to demonstrate command fragment completion. So when a user types `git ` (note the space), it gets to return an array of possible subcommands that can come after `git `. What you did would produce something like `git git init` if there's a space after the entered command